compiler/hsSyn/HsUtils.hs

```diff
@@ -371,14 +371,15 @@ mkHsStringPrimLit fs
 
 -------------
 userHsLTyVarBndrs :: SrcSpan -> [Located (IdP (GhcPass p))]
+                  -> [HasBraces]
                   -> [LHsTyVarBndr (GhcPass p)]
 -- Caller sets location
-userHsLTyVarBndrs loc bndrs = [ L loc (UserTyVar noExt v) | v <- bndrs ]
+userHsLTyVarBndrs loc bndrs braces = [ L loc (UserTyVar noExt v brace) | v <- bndrs, brace <- braces ]
```

Comment:
This doesn't look right. Did you meant to `zip` these lists instead of making a nested loop (each-with-each)?